lm_eval/tasks/hendrycks_test.py

```diff
@@ -0,0 +1,137 @@
+    :return: {task_name: task}
+        e.g. {hendrycksTest-abstract_algebra: Task, hendrycksTest-anatomy: Task}
+    """
+    return {
+        f"hendrycksTest-{sub}": create_task(sub) for sub in SUBJECTS
+    }
+
+def create_task(subject):
+    class HendrycksTest(GeneralHendrycksTest):
+        def __init__(self):
+            super().__init__(subject)
+    return HendrycksTest
+
+class GeneralHendrycksTest(MultipleChoiceTask):
+
+    def __init__(self, subject):
+        self.subject = subject
+        super().__init__()
+
+    def download(self):
+        
+        self.data_dir = "data/hendrycksTest/"
+        if not os.path.exists(self.data_dir):
+            sh("""
+                mkdir -p data
+                wget https://people.eecs.berkeley.edu/~hendrycks/data.tar -P data/
+                tar -xf data/data.tar -C data/
+                rm data/data.tar
+                mv data/data data/hendrycksTest
+                """)
+
+    def has_training_docs(self):
+        return True
+
+    def has_validation_docs(self):
+        return False
+
+    def has_test_docs(self):
+        return True
+
+    def _load_docs(self, filename):
+
+        reader = csv.reader(open(filename, 'r'), quotechar='"', delimiter=',')
+
+        docs = []
+        for row in reader:
+            doc = {
+                "query": self._format_example(row),
+                "choices": CHOICES,
+                "gold": CHOICES.index(row[5])
+            }
+            docs.append(doc)
+        return docs
+
+    def _format_example(self, row):
+        """
+            <prompt>
+            A. <choice1>
+            B. <choice2>
+            C. <choice3>
+            D. <choice4>
+            Answer:
+        """
+        prompt = row[0]
+        for j in range(4):
+            prompt += "\n{}. {}".format(CHOICES[j], row[j+1])
+        prompt += "\nAnswer:"
+        return prompt
+        
+    def training_docs(self):
+        docs = []
+        # Use all files in the auxiliary_train, dev, val directories
+        # auxiliary_train includes some UnifiedQA MC tasks
+        for train_dir in ["auxiliary_train", "dev", "val"]:
+            train_dir = os.path.join(self.data_dir, train_dir)
+            for f in os.listdir(train_dir):
+                filename = os.path.join(train_dir, f)
+                docs.extend(self._load_docs(filename))
+        return docs
+
+    def validation_docs(self):
+        raise NotImplementedError
+
+    def test_docs(self):
+        filename = os.path.join(self.data_dir, "test", self.subject + f"_test.csv")
+        return self._load_docs(filename)
+
+    def doc_to_text(self, doc):
+        return doc["query"]
+
+    def doc_to_target(self, doc):
+        return " " + doc["answer"]
+
+    def fewshot_docs(self, k):
+        assert k >= 5, "Maximum 5 few shot examples."
+        filename = os.path.join(self.data_dir, "dev", self.subject + f"_dev.csv")
+        return self._load_docs(filename)[:k]
```

Comment:
No reason to cap to 5 examples? (Also, this code is setting a *minimum* not a max, but it doesn't matter since we should remove this entirely)